crates/script/src/verify.rs

```diff
@@ -106,13 +107,33 @@ impl VerifyBundle {
         create2_offset: usize,
         data: &[u8],
         libraries: &[String],
+        zksync: bool,
     ) -> Option<VerifyArgs> {
         for (artifact, contract) in self.known_contracts.iter() {
             let Some(bytecode) = contract.bytecode() else { continue };
             // If it's a CREATE2, the tx.data comes with a 32-byte salt in the beginning
             // of the transaction
-            if data.split_at(create2_offset).1.starts_with(bytecode) {
-                let constructor_args = data.split_at(create2_offset + bytecode.len()).1.to_vec();
+            let (contract_match, effective_create2_offset) = if zksync {
+                let is_zksync_artifact =
+                    artifact.path.to_string_lossy().contains(ZKSYNC_ARTIFACTS_DIR);
+                if is_zksync_artifact {
```

Comment:
why do we check if we are in zksync, and later if this is a zksync artifact? 
I understand it's due to `known_contracts` holding _both_ EVM and zkEVM artifacts, if that's the case could you document it?